[te] Disable reductions by default

bertmaher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #44152 Fix frac in CUDA fuser
* #44160 Prevent the TE fuser from getting datatypes it can't handle
* #44157 [te] Fix casting of unsigned char, and abs(int)
* **#44122 [te] Disable reductions by default**

Differential Revision: [D23504769](https://our.internmc.facebook.com/intern/diff/D23504769)

bertmaher committed:
[te] Disable reductions by default

[ghstack-poisoned]

eellison commented:
LGTM - this will conflict with https://github.com/pytorch/pytorch/pull/44083/ - mind waiting till i land ?

bertmaher commented:
> LGTM - this will conflict with #44083 - mind waiting till i land ?

Nope, go right ahead, and I'll rebase after yours is in!

bertmaher committed:
Update on "[te] Disable reductions by default"


Differential Revision: [D23504769](https://our.internmc.facebook.com/intern/diff/D23504769)

[ghstack-poisoned]

bertmaher committed:
Update on "[te] Disable reductions by default"


Differential Revision: [D23504769](https://our.internmc.facebook.com/intern/diff/D23504769)

[ghstack-poisoned]

bertmaher committed:
Update on "[te] Disable reductions by default"


Differential Revision: [D23504769](https://our.internmc.facebook.com/intern/diff/D23504769)

[ghstack-poisoned]
